internal/arduino/cores/packagemanager/package_manager.go

```diff
@@ -634,20 +633,14 @@ func (pme *Explorer) GetInstalledPlatformRelease(platform *cores.Platform) *core
 	for _, candidate := range releases[1:] {
 		candidateIsManaged := pme.IsManagedPlatformRelease(candidate)
 		debug("candidate", candidate)
-		// TODO: Disentangle this algorithm and make it more straightforward
-		if bestIsManaged == candidateIsManaged {
-			if best.IsIDEBundled == candidate.IsIDEBundled {
-				if candidate.Version.GreaterThan(best.Version) {
-					best = candidate
-				}
-			}
-			if best.IsIDEBundled && !candidate.IsIDEBundled {
-				best = candidate
-			}
+		if !candidateIsManaged {
+			continue
 		}
-		if !bestIsManaged && candidateIsManaged {
+		if !bestIsManaged {
 			best = candidate
 			bestIsManaged = true
+		} else if candidate.Version.GreaterThan(best.Version) {
+			best = candidate
 		}
```

Comment:
Nice catch, although extremely rare, it can happen only if someone is manually installing a core using this folder structure:
`ARCH/VERSION/*`

Fixed by https://github.com/arduino/arduino-cli/pull/2424/commits/4911d96f2bbfa5830a2858662256f294c5669e9f